ui/components/multichain/app-header/app-header.test.js

```diff
@@ -175,4 +185,48 @@ describe('App Header', () => {
       expect(connectionPickerButton).not.toBeInTheDocument();
     });
   });
+
+  describe('network picker', () => {
+    it('shows custom rpc if it has the same chainId as a default network', () => {
+      const mockProviderConfig = {
+        id: 'custom-rpc-localhost',
+        type: 'rpc',
+        ticker: 'ETH',
+        chainId: '0x1',
+        rpcUrl: 'https://localhost:8545',
+        nickname: 'Localhost',
+      };
+
+      const mockNetworkConfigurations = {
+        [mockProviderConfig.id]: mockProviderConfig,
+      };
```

Comment:
Nit: actually, the suggestion was not applied 🙈 this one is mainly to group mock declarations altogether (like in "shows rpc url as nickname if there isn't a nickname set")

```suggestion
      const mockNetworkConfigurations = {
        [mockProviderConfig.id]: mockProviderConfig,
      };

```